server/sonar-web/src/main/js/components/SelectList/SelectList.tsx

```diff
@@ -0,0 +1,115 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import * as React from 'react';
+import SelectListListContainer from './SelectListListContainer';
+import { translate } from '../../helpers/l10n';
+import SearchBox from '../controls/SearchBox';
+
+interface Props {
+  onSearch: (query: string, selected: string) => Promise<void>;
+  onSelect: (id: number) => Promise<void>;
+  onUnselect: (id: number) => Promise<void>;
+  elements: Array<{ id: number; name: string; selected: boolean }>;
+}
+
+interface State {
+  activeTab: number;
+  query: string;
+}
+
+export default class SelectList extends React.PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      activeTab: 0,
+      query: ''
+    };
+  }
+
+  getSelectedValueFromTabIndex(tabIndex: number) {
+    if (tabIndex === 0) {
+      return 'selected';
+    }
+    return tabIndex === 1 ? 'deselected' : 'all';
+  }
+
+  changeTab(tabIndex: number) {
+    this.setState({ activeTab: tabIndex });
+    this.props.onSearch(this.state.query, this.getSelectedValueFromTabIndex(tabIndex));
+  }
+
+  handleQueryChange = (query: string) => {
+    this.setState({ query });
+    this.props.onSearch(query, this.getSelectedValueFromTabIndex(this.state.activeTab));
+  };
+
+  handleSelect = (id: number) => {
+    this.props.onSelect(id);
+  };
+
+  handleUnselect = (id: number) => {
+    this.props.onUnselect(id);
+  };
+
+  render() {
+    const { activeTab } = this.state;
+
+    return (
+      <div>
+        <div className="button-group pull-left spacer-right">
+          <button
+            className={activeTab === 0 ? 'button-active' : undefined}
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Define a constant instead of duplicating this literal 3 times. ([typescript:S1192](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS1192))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&pullRequest=3152&issues=AWIj7wLI7DpOGbPQ2n0_&open=AWIj7wLI7DpOGbPQ2n0_)

[ISSUE_KEY]: AWIj7wLI7DpOGbPQ2n0_